wa/workloads/hwuitest/__init__.py

```diff
@@ -0,0 +1,113 @@
+#    Copyright 2013-2018 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#pylint: disable=E1101,W0201
+
+import os
+import re
+from collections import defaultdict
+
+from wa import Workload, Parameter, Executable
+from wa.utils.types import caseless_string
+
+
+BINARY = "hwuitest"
+IGNORED_METRICS = ["Stats since", "Total frames rendered"]
+
+
+class HWUITest(Workload):
+
+    name = 'hwuitest'
+    description = """
+    Tests UI rendering latency on Android devices.
+
+    The binary for this workload is built as part of AOSP's 
+    frameworks/base/libs/hwui component.
+    """
+    supported_platforms = ['android']
+
+    parameters = [
+        Parameter('test', kind=caseless_string, default="shadowgrid",
+                  allowed_values=["shadowgrid", "rectgrid", "oval"],
+                  description="""
+                  The test to run:
+
+                     - ``'shadowgrid'``: creates a grid of rounded rects that
+                       cast shadows, high CPU & GPU load
+                     - ``'rectgrid'``: creates a grid of 1x1 rects
+                     - ``'oval'``: draws 1 oval
+                  """),
+        Parameter('loops', kind=int, default=3,
+                  description="The number of test iterations."),
+        Parameter('frames', kind=int, default=150,
+                  description="The number of frames to run the test over."),
+    ]
+
+    def setup(self, context):
+        host_exe = context.resolver.get(Executable(self,
+                                                   self.target.abi,
+                                                   BINARY))
+        self.target_exe = self.target.install(host_exe)
```

Comment:
Re-factor the executable deployment / cleanup to avoid duplicate installations with multiple iterations?